pkg/network/dhcp/dhcp.go

```diff
@@ -1,25 +1,29 @@
 package dhcp
 
 import (
+	"context"
 	"os/exec"
-	"time"
 
 	"github.com/rs/zerolog/log"
-	"github.com/threefoldtech/zos/pkg/network/ifaceutil"
-	"github.com/vishvananda/netlink"
 )
 
-// Probe will do a dhcp request on the interface inf
-// if the interface gets a lease from the dhcp server, dhcpProbe return true and a nil error
-// if something unexpected happens a non nil error is return
-// if the interface didn't receive an lease, false and a nil error is returns
-func Probe(inf string, family int) (bool, error) {
-	link, err := netlink.LinkByName(inf)
-	if err != nil {
-		return false, err
+// Probe is used to do some DHCP request on a interface
+type Probe struct {
+	ctx context.Context
+	cmd *exec.Cmd
+}
+
+// NewPrope returns a Probe
+func NewPrope(ctx context.Context) *Probe {
```

Comment:
I have read somewhere (can't find the article) that context are supposedly only used for "request" based context. In other words, you should not store the context as an attribute anywhere. Instead it must be passed to a method (not a constructor)

Hence this should be:
```go
func NewPrope() *Prope {}
func (d *Probe) Start(ctx context.Context) erro {}
```
instead.

*Note*: I actually don't see where this context is used anyway!

Reply:
> Note: I actually don't see where this context is used anyway!

Indeed :-P I'll just remove it